src/DGtal/geometry/surfaces/DigitalSurfaceConvolver.ih

```diff
@@ -1096,15 +1096,18 @@ DGtal::DigitalSurfaceConvolver< Functor, KernelFunctor, KSpace, DigitalKernel, 2
               {
                 for( KernelConstIterator itm = myKernelMask->first; itm != myKernelMask->second; ++itm )
                   {
-                    myKSpace.sSetKCoords( shiftedSpel, myKSpace.sKCoords( myKSpace.sSpel( *itm )) + shiftInnerSpel );
+                    if( myKSpace.cIsInside( myKSpace.sKCoords( myKSpace.sSpel( *itm )) + shiftInnerSpel ))
+                    {
+                      myKSpace.sSetKCoords( shiftedSpel, myKSpace.sKCoords( myKSpace.sSpel( *itm )) + shiftInnerSpel );
```

Comment:
Yes, I think so ... maybe it could trigger an error at https://github.com/jlevallois/DGtal/blob/dev_periodicKhalimsky/src/DGtal/geometry/surfaces/DigitalSurfaceConvolver.ih#L1205 and https://github.com/jlevallois/DGtal/blob/dev_periodicKhalimsky/src/DGtal/geometry/surfaces/DigitalSurfaceConvolver.ih#L1476 and other similar lines but I don't think so.

So, it is not very important ;)